### DISCUSSION TITLE
Does/will this library support reinforcement learning such as Deep Q-Learning?

### DISCUSSION THREAD

**Discussion by xatok** [Kotlin/kotlindl#510] [Q&A] [answered]
I'm interested in using the Deep Q-Learning for games using Kotlin and I'd definitely use this library if it supports it, but unfortunately, it seems that this library is more focused on image recognition (correct me if I'm wrong).

Should I then use other libraries like Deeplearning4j? Or Kotlindl supports it?

**Comment by juliabeliaeva** [comment, marked as answer]
Hi, thank you for your inquiry. The library is meant to be general-purpose, so you can train a model which receives some float array and outputs another float array, and interpret the result however you want. But there is no support specifically for reinforcement learning, and no tutorials/examples in this area. If Deeplearning4j works for reinforcement learning out of the box, than it is probably the better option.
